trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid.F

```diff
@@ -0,0 +1,350 @@
+! This module defines and instantiates objects
+! for a hybrid persistence levelpool type
+! reservoir. The hybrid reservoir struct
+! inherits input and output stucts from the
+! reservoir base module and calls instantiation
+! of these into sub-objects. The hybrid
+! reservoir struct also points to structs for
+! hybrid parameters and state and calls
+! instantiation of these into sub-objects.
+! A pointer to a levelpool reservoir object
+! is also held in state, and this module
+! instantiates that levelpool object. There
+! is also a subroutine to run hybrid release
+! that is derived from the reservoir base
+! struct interface to run release.
+
+! FIXME: Change value for WCOS
+#define log_warning 6
```

Comment:
hmm, this is border line. I say we dont use CPP to do variable subsitution. We dont currently do this anywhere that I can think if, we just have ifdefs.

Reply:
As per my comment on the previous macro disscussion, this may not be the appropriate way, but I can see where all the I/O file handles that require special mapping on WCOSS (i.e. stdout/stderror) could be defined by macros, making the code much more concise, i.e.:
```#ifdef WCOSS
#define STDOUT 15
#else
#define STDOUT 5

write(STDOUT, *) "write to std out" 
```
Fortran 2003 does come with an io module that tryies to encapsulate this at the compiler level, and you can overwrite the module with specific parameters, but that seems to be a lot of explicit coding required to support different platforms.  This is what pre-processing is designed for. I don't see a large issue is using these features for build/environment/system management whether or not is fortranic.  For these cases, it is a tool, not code.  For the case of application parameter substitution, a module is good.

Reply:
I agree this is reasonable for WCOSS, as that's a compile configuration. It's not appropriate for invariants, i.e. π, days-in-year, etc. 

One other thing to note here is that 5 is STDOUT by convention but **not** by the language standard. `iso_fortran_env`  defines `input_unit`, `output_unit`, and `error_unit` and should be used where supported (all modern compilers).

Reply:
> One other thing to note here is that 5 is STDOUT by convention but **not** by the language standard.

This is why many system applications use macros and "configure" the build based on system parameters.  To use `iso_fortran_env` do you have to override the module for each platform you intend to support?  Or is it the responsibility of the system to "configure" the compiler with their specific details of the system so that when used by an application it is correct?  I don't like the idea of assuming something else is properly configured, nor do I like the concept of re-writing a lot of code to simply do variable substitution configuring.

Reply:
`iso_fortran_env` requires no user configuration; one of its primary purposes is to provide standard symbolic constants for these things that may be different between platforms.

Reply:
Hmm...so how could it be used to differentiate the strict requirement on WCOSS that stdout writes to a specific file unit and other systems don't care?

Reply:
You still need the #ifdef, appropriate because of the compile-time configuration; my point was that using a hardcoded 5 for all other systems wasn't actually standards-compliant.
 

e.g.:
```
use iso_fortran_env, only: output_unit
...
#ifdef WCOSS
#define output_unit 15
#endif
write(output_unit, *) "write to std out" 
```

It's entirely possible that WCOSS has a custom `iso_fortran_env` that already defines `output_unit` as 15, but I'd have to defer that to someone more familiar with that environment...

Reply:
Thanks for the clarification.  The documentation I was browsing didn't give me the correct impression.

Reply:
@jdmattern-noaa fixing. Changing to match logging in module_HYDRO_drv.F with #ifndef. 

#ifndef NCEP_WCOSS
    integer, parameter :: log_warning = 6
#else
    integer, parameter :: log_warning = 78
#endif